### ISSUE TITLE
Freeze on start during liquibase step

### ISSUE THREAD

**Issue by demtnt** [quarkusio/quarkus#42645]
### Describe the bug

There is simple micro service
```  
    implementation(enforcedPlatform("io.quarkus.platform:quarkus-amazon-services-bom:${quarkusVersion3}"))
    implementation(enforcedPlatform("io.quarkus.platform:quarkus-bom:${quarkusVersion3}"))
    implementation("io.quarkus:quarkus-container-image-docker")
    implementation("io.quarkus:quarkus-config-yaml")
    implementation("io.quarkus:quarkus-arc")
    implementation("io.quarkus:quarkus-jdbc-postgresql")
    implementation("software.amazon.awssdk:apache-client")
    implementation("io.quarkus:quarkus-liquibase")
    implementation("io.smallrye.reactive:smallrye-mutiny-vertx-web-client")
    implementation("io.quarkus:quarkus-resteasy-reactive-jackson")
    implementation("com.fasterxml.jackson.module:jackson-module-kotlin")
    implementation("io.quarkus:quarkus-resteasy-reactive")
    implementation("io.quarkus:quarkus-hibernate-validator")
    implementation("io.quarkus:quarkus-swagger-ui")
    implementation("io.quarkus:quarkus-smallrye-openapi")
    implementation("io.quarkus:quarkus-smallrye-health")
    implementation("io.quarkus:quarkus-opentelemetry")
    implementation("io.quarkus:quarkus-micrometer-registry-prometheus")
```
After migrating to 3.13.2(from version 3.12.3) program freezing during liquidate step. 

last logs are:
```
07:56:46 INFO   [li.database] (main) Set default schema name to public
07:57:13 FINE   [li.configuration] (main) No configuration value for liquibase.liquibaseSchemaName aka liquibase.schemaName found
07:58:31 FINE   [li.configuration] (main) No configuration value for liquibase.supportPropertyEscaping aka enableEscaping found
07:58:31 FINE   [li.configuration] (main) Configuration liquibase.supportPropertyEscaping is using the default value of false
07:58:56 FINE   [li.servicelocator] (main) Loaded liquibase.lockservice.LockService instance liquibase.lockservice.LockServiceImpl
07:58:56 FINE   [li.servicelocator] (main) Loaded liquibase.lockservice.LockService instance liquibase.lockservice.MockLockService
07:58:56 FINE   [li.servicelocator] (main) Loaded liquibase.lockservice.LockService instance liquibase.lockservice.OfflineLockService
07:58:56 FINE   [li.servicelocator] (main) Loaded liquibase.lockservice.LockService instance liquibase.lockservice.StandardLockService
07:59:22 FINE   [li.configuration] (main) No configuration value for liquibase.changelogLockWaitTimeInMinutes aka liquibase.changeLogLockWaitTimeInMinutes found
07:59:22 FINE   [li.configuration] (main) Configuration liquibase.changelogLockWaitTimeInMinutes is using the default value of 5
08:00:05 DEBUG  [io.qu.ru.Application] (SIGTERM handler) Received signed 15, shutting down
```
 If we set  `quarkus.liquibase.migrate-at-start: false` - everything works fine.

liquibase file located in default location `db/changeLog.xml` contains primitive changeset.

Probably it could be connected with: https://github.com/quarkusio/quarkus/pull/42597 (Downgrade commons-lang3 to 3.14.0 to avoid entropy issues)

### Expected behavior

Liquibase migration in native image works as expected.

### Output of `uname -a` or `ver`

Linux jenkins-node2 5.14.0-427.24.1.el9_4.x86_64 #1 SMP PREEMPT_DYNAMIC Mon Jul 8 17:47:19 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

12:17:52  openjdk version "21.0.3" 2024-04-16 LTS
12:17:52  OpenJDK Runtime Environment (build 21.0.3+12-LTS)
12:17:52  OpenJDK 64-Bit Server VM (build 21.0.3+12-LTS, mixed mode, sharing)

### Quarkus version or git rev

3.13.2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

12:17:52  Gradle 8.10
12:17:52  ------------------------------------------------------------
12:17:52  
12:17:52  Build time:    2024-08-14 11:07:45 UTC
12:17:52  Revision:      fef2edbed8af1022cefaf44d4c0514c5f89d7b78
12:17:52  
12:17:52  Kotlin:        1.9.24
12:17:52  Groovy:        3.0.22
12:17:52  Ant:           Apache Ant(TM) version 1.10.14 compiled on August 16 2023
12:17:52  Launcher JVM:  21.0.3 (BellSoft 21.0.3+12-LTS)
12:17:52  Daemon JVM:    /opt/liberica-jdk21 (no JDK specified, using current Java home)
12:17:52  OS:            Linux 5.14.0-427.24.1.el9_4.x86_64 amd64

**Comment by gsmet** [comment]
I think it could be due to this issue: https://github.com/quarkusio/quarkus/pull/42597 .

Try downgrading commons-lang3 to 3.14.0 in your app and see how it goes.

I'm currently deploying 3.13.3 with the downgrade.

**Comment by gsmet** [comment]
Ah sorry, didn't see you also connected the dots. Give it a try and report back or wait a few hours for the 3.13.3 announcement.

**Comment by demtnt** [comment]
```
configurations.all {
    resolutionStrategy.dependencySubstitution {
        substitute(module("org.apache.commons:commons-lang3"))
            .using(module("org.apache.commons:commons-lang3:3.14.0"))
    }
}
```
Helps me

**Comment by gsmet** [comment]
@demtnt 3.13.3 is available on Maven Central so I will close this. Please reopen if you still experience the issue.

Thanks for the report.

**Comment by filipelautert** [comment]
This is being fixed for Liquibase 4.29.2 next week: https://github.com/liquibase/liquibase/pull/6179

**Comment by gsmet** [comment]
Thanks for the heads up. We will wait for the next version and maybe wait some more for other projects to catch up before updating commons-lang3 again.
